lib/statsd.rb

```diff
@@ -128,8 +165,24 @@ def time(stat, sample_rate=1)
   # for the stat.
   def histogram(stat, value, sample_rate=1); send stat, value, HISTOGRAM_TYPE, sample_rate end
 
-  private
+  def enable_buffering!(buffer_size = nil)
+    return if @buffering
+    @shards.map! { |client| BufferWrapper.new(client, buffer_size) }
+    @buffering = true
+  end
+
+  def disable_buffering!
+    return unless @buffering
+    @shards.map! { |client| client.base_client }
+    @buffering = false
+  end
```

Comment:
Maybe this should also call `flush_all!`? I suppose that could be left to the caller; perhaps a comment explaining that would be good.